inject-java/src/test/groovy/io/micronaut/inject/annotation/BeanDefinitionAnnotationMetadataSpec.groovy

```diff
@@ -282,7 +282,7 @@ interface Foo {
         BeanDefinition definition = classLoader.loadClass('test.$Test$Foo0Definition').newInstance()
         expect:
         definition != null
-        definition.hasStereotype(AnnotationUtil.SINGLETON)
+        definition.hasAnnotation(AnnotationUtil.SINGLETON)
```

Comment:
If singleton isn't inherited the factory definition shouldn't have the annotation, right?

Reply:
I think in this case a factory should be considered "extending" the return type, otherwise things like `@Requires` couldn't be changed via a factory

Reply:
That is the current behavior. I haven't changed anything there. It is imo debatable whether annotations on the type returned should be taken into account at all.

Reply:
Right I think this is one of the areas that should be addressed as part of this. If you prefer to resolve it after this PR that is OK with me. Please bring it up at the next meeting. In my opinion I don't think any annotations should be taken into account.

Reply:
I agree with you :)